src/uct/base/uct_iov.inl

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_IOV_INL_
+#define UCT_IOV_INL_
+
+#include <uct/api/uct.h>
+#include <ucs/sys/math.h>
+#include <ucs/debug/assert.h>
+
+#include <ucs/sys/iovec.h>
+#include <ucs/sys/iovec.inl>
+
+
+/**
+ * Calculates the total length of the particular UCT IOV data buffer.
+ *
+ * @param [in]     iov             Pointer to the UCT IOV element.
+ *
+ * @return The length of the UCT IOV data buffer.
+ * @note Currently has no support for the strides. If the strides are
+ *       supported, it should be like: length + ((count - 1) * stride)
+ */
+static UCS_F_ALWAYS_INLINE
+size_t uct_iov_get_length(const uct_iov_t *iov)
+{
+    return iov->count * iov->length;
+}
+
+/**
+ * Returns the particular UCT IOV data buffer.
+ *
+ * @param [in]     iov             The pointer to the UCT IOV element.
+ *
+ * @return The UCT IOV data buffer.
+ */
+static UCS_F_ALWAYS_INLINE
+void *uct_iov_get_buffer(const uct_iov_t *iov)
+{
+    return iov->buffer;
+}
+
+/**
+ * Calculates the total length of the UCT IOV array buffers.
+ *
+ * @param [in]     iov             Pointer to the array of UCT IOVs.
+ * @param [in]     iovcnt          Number of the elements in the array of UCT IOVs.
+ *
+ * @return The total length of the array of UCT IOVs.
+ */
+static UCS_F_ALWAYS_INLINE
+size_t uct_iov_total_length(const uct_iov_t *iov, size_t iovcnt)
+{
+    size_t iov_it, total_length = 0;
+
+    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+        total_length += uct_iov_get_length(&iov[iov_it]);
+    }
+
+    return total_length;
+}
```

Comment:
can we add generic ucs function to calculate iov total length? like the converter?